platform/linux-generic/odp_rwlock.c

```diff
@@ -33,9 +33,14 @@ void odp_rwlock_read_lock(odp_rwlock_t *rwlock)
 
 int odp_rwlock_read_trylock(odp_rwlock_t *rwlock)
 {
-	uint32_t zero = 0;
+	uint32_t cnt = odp_atomic_load_u32(&rwlock->cnt);
+
+	while (cnt != (uint32_t)-1) {
```

Comment:
Good point, and a reasonable one. Trylock says the caller doesn't want to wait for the lock to be released, but given that we're now allowing multiple readers theres the possibility that CAS operations may need to be retried as they all try to increment in parallel. So I withdraw this comment.